base/applications/rapps/appview.cpp

```diff
@@ -970,16 +973,89 @@ CAppInfoDisplay::~CAppInfoDisplay()
 
 // **** CAppsListView ****
 
+struct CAsyncLoadIcon {
+    CAsyncLoadIcon *pNext;
+    HWND hAppsList;
+    CAppInfo *AppInfo; // Only used to find the item in the list, do not access on background thread
+    UINT TaskId;
+    bool Parse;
+    WCHAR Location[1];
```

Comment:
```suggestion
    WCHAR Location[ANYSIZE_ARRAY];
```
or some dynamic buffer?

Reply:
I want to do as little as possible on the UI thread so not two allocations.